tests/unit_tests/test_invalid_build.py

```diff
@@ -0,0 +1,48 @@
+import pytest
+
+from magniv.build.build import build
+from tests.unit_tests.fixtures.test_files import (
+    INVALID_KWARG_FILE,
+    INVALID_TASK_KEY_FILE,
+    INVALID_TRIGGER_FILE,
+)
+
+# TODO: merge with TestBuild upon refactor there
+
+
```

Comment:
BTW, what makes `@pytest.fixture` better than just doing something like:

```python
class TestBuildBase:
  def folder():
     file = self.get_file()
   ...
   
   def get_file(self):
       return FILE_STR
```

Reply:
EDIT: Seems like this is important for using `build`?

Reply:
Yeah, the crucial aspect of it here is [tmpdir](https://docs.pytest.org/en/7.1.x/reference/reference.html#std-fixture-tmpdir) - it's totally nonobvious what's going on there but basically it's ensuring that every test runs in an isolated, temporary directory. I had actually never seen this before looking at this code but it's cool. Will add some comments to clarify the behavior

Also just noticed by reading those docs again that it looks like `tmp_path` is preferred now...

Reply:
Okay updated to use `tmp_path`! but same idea applies. clarified this in the docstring, too